src/framework/PreferenceStore/PreferenceStoreContext.tsx

```diff
@@ -0,0 +1,185 @@
+  orgIdentifier?: string
+  userId?: string
+}
+
+export interface PreferenceStoreContextProps<T> {
+  preference: T
+  setPreference: (value: T) => void
+  clearPreference: () => void
+  updatePreferenceStore: (data: PreferenceStoreStateProps) => void
+}
+
+export const PREFERENCES_TOP_LEVEL_KEY = 'preferences'
+
+export const PreferenceStoreContext = React.createContext<PreferenceStoreProps<any>>({
+  set: () => void 0,
+  get: () => void 0,
+  clear: () => void 0,
+  updatePreferenceStore: () => void 0
+})
+
+export function usePreferenceStore<T>(
+  scope: PreferenceScope,
+  entity: string,
+  options: PreferenceStoreOptions = {}
+): PreferenceStoreContextProps<T> {
+  const { get, set, clear, updatePreferenceStore } = React.useContext(PreferenceStoreContext)
+
+  const preference = get(scope, entity, options)
+  const setPreference = set.bind(null, scope, entity, options)
+  const clearPreference = clear.bind(null, scope, entity, options)
+
+  return { preference, setPreference, clearPreference, updatePreferenceStore }
+}
+
+const checkAccess = (scope: PreferenceScope, contextArr: (string | undefined)[]): void => {
+  if (!contextArr || contextArr?.some(val => val === undefined)) {
+    throw new Error(`Access to "${scope}" scope is not available in the current context.`)
+  }
+}
+
+const getKey = (arr: (string | undefined)[], entity: string): string => {
+  return [...arr, entity].join('/')
+}
+
+export const PreferenceStoreProvider: React.FC = (props: React.PropsWithChildren<unknown>) => {
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
+  const [state, setState] = React.useState<PreferenceStoreStateProps>({
+    currentUserInfo: {}
+  })
+  const [currentPreferences, setPreferences] = useLocalStorage<Record<string, unknown>>(PREFERENCES_TOP_LEVEL_KEY, {})
+  const userId = state.currentUserInfo?.email
+  const [scopeToKeyMap, setScopeToKeyMap] = React.useState({
+    [PreferenceScope.USER]: [userId],
+    [PreferenceScope.ACCOUNT]: [accountId],
+    [PreferenceScope.ORG]: [accountId, orgIdentifier],
+    [PreferenceScope.PROJECT]: [accountId, orgIdentifier, projectIdentifier],
+    [PreferenceScope.MACHINE]: []
+  })
+
+  useEffect(() => {
+    setScopeToKeyMap({
+      [PreferenceScope.USER]: [userId],
+      [PreferenceScope.ACCOUNT]: [accountId],
+      [PreferenceScope.ORG]: [accountId, orgIdentifier],
+      [PreferenceScope.PROJECT]: [accountId, orgIdentifier, projectIdentifier],
+      [PreferenceScope.MACHINE]: []
+    })
+  }, [accountId, orgIdentifier, projectIdentifier, userId])
+
+  const setPreference = (key: string, value: unknown, options?: PreferenceStoreOptions): void => {
+    if (options?.fromBackend) {
+      // TODO: ENHANCEMENT: call backend to set
+    } else {
+      const newPreferences = { ...currentPreferences, [key]: value }
+      setPreferences(newPreferences)
+    }
+  }
+
+  const getPreference = (key: string, options?: PreferenceStoreOptions): any => {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232308)
```suggestion
  const getPreference = (key: string, options?: PreferenceStoreOptions): unknown => {
```